src/tiled/objectselectiontool.cpp

```diff
@@ -1431,20 +1433,45 @@ void ObjectSelectionTool::saveSelectionState()
 void ObjectSelectionTool::refreshCursor()
 {
     Qt::CursorShape cursorShape = Qt::ArrowCursor;
-
+    const MapObject *prevObject = nullptr;
     switch (mAction) {
     case NoAction: {
         const bool hasSelection = !mapScene()->selectedObjectItems().isEmpty();
 
         if ((mHoveredObjectItem || ((mModifiers & Qt::AltModifier) && hasSelection && !mHoveredHandle)) &&
                 !(mModifiers & Qt::ShiftModifier)) {
             cursorShape = Qt::SizeAllCursor;
+            MapObject *currentObject = mHoveredObjectItem->mapObject();
```

Comment:
This may crash since there is no guarantee `mHoveredObjectItem` isn't null at this point.